### PR TITLE
Support for force-add override syntax (++key=value)

### LINKED ISSUE #1049: Add OR Override-if-exists
# 🚀 Feature Request

Add option to "Add-OR-Override-if-exists" to command line params

e.g.
```
python myscript.py !gpus=2 # gpus will be set to two, disregarding whether it existed in the original config file or not
```

## Motivation

Hydra offers few options to handle params from command line:
```
python myscript.py gpus=2 # override gpus (must exists in config files/structured config earlier!)
```
```
python myscript.py ~gpus # removes gpus from configuration 
```
```
python myscript.py +gpus=2 # add gpus (cannot exists in config files/structured config earlier!)
```

Note +gpus will raise an error if it was already present in the configuration.

So the case, when we want to have gpus set to 2, disregarding whether it was present in yaml file or now, is missing.

### PR DESCRIPTION
**Pull request by omry**
Closes #1049

### CODE REVIEW

File: hydra/_internal/defaults_list.py [hunk 1]
```diff
@@ -69,7 +69,12 @@ def __init__(self, repo: IConfigRepository, overrides_list: List[Override]) -> N
             if is_dict or not is_group:
                 self.config_overrides.append(override)
             else:
-                if override.is_delete():
+                if override.is_force_add():
```
Comment (odelalleau) [inline hydra/_internal/defaults_list.py:72]: Not actually related to this PR, but you could gain some indentation back by using an`elif` here

Comment (omry) [reply hydra/_internal/defaults_list.py]: yup, done.

**Review by odelalleau — CHANGES_REQUESTED** [review]
Just one thing, I suspect that `get_key_element()` in`types.py` also needs to be updated, could you have a look and add a test for it?

**Comment by omry** [comment]
> Just one thing, I suspect that `get_key_element()` in`types.py` also needs to be updated, could you have a look and add a test for it?

nice catch.

**Review by odelalleau — APPROVED** [review]
All good once CI gives the green light!

**Comment by kenahoo** [comment]
I tested this for my use case, it seems to work well.  Thanks for looking at it.

**Comment by Gpapi13** [comment]
- [ ]

**Pull request merged**